pep-0612.rst

```diff
@@ -77,49 +75,79 @@ arguments.
 Without the ability to define dependencies between the parameters of different
 callable types, there is no way, at present, to make ``add_logging`` compatible
 with all functions, while still preserving the enforcement of the parameters of
-the decorated function. 
+the decorated function.
 
-With the addition of the ``ParameterSpecification`` variables proposed by this
+With the addition of the ``ParamSpec`` variables proposed by this
 PEP, we can rewrite the previous example in a way that keeps the flexibility of
 the decorator and the parameter enforcement of the decorated function.
 
 .. code-block::
 
-   from typing import Awaitable, Callable, ParameterSpecification, TypeVar
+   from typing import Awaitable, Callable, ParamSpec, TypeVar
 
-   Ps = ParameterSpecification("Ps")
+   P = ParamSpec("P")
    R = TypeVar("R")
 
-   def add_logging(f: Callable[Ps, R]) -> Callable[Ps, Awaitable[R]]:
-       async def inner(*args: Ps.args, **kwargs: Ps.kwargs) -> R:
+   def add_logging(f: Callable[P, R]) -> Callable[P, Awaitable[R]]:
+       async def inner(*args: P.args, **kwargs: P.kwargs) -> R:
            await log_to_database()
            return f(*args, **kwargs)
-
        return inner
 
    @add_logging
    def foo(x: int, y: str) -> int:
        return x + 7
 
    await foo(1, "A")
-   await foo("B", 2) # Incompatible parameter type: 
-                     # Expected `int` for 1st anonymous parameter to call `foo` 
+   await foo("B", 2) # Incompatible parameter type:
+                     # Expected `int` for 1st anonymous parameter to call `foo`
                      # but got `str`
 
+With the further addition of the ``Concatenate`` operator, we can also express
+the types of more complex decorators which add, remove, or modify a finite
+number of  positional arguments from decorated functions.
+
+.. code-block::
+
+   from typing.type_variable_operators import Concatenate
+
+   def add(f: Callable[P, str]) -> Callable[Concatenate[str, bool, P], int]:
+       def inner(x: str, y: bool, *args: P.args, **kwargs: P.kwargs) -> int:
+           use(x, y)
```

Comment:
nit: define `use`?

Reply:
I think it might be better to just not have that, or replace it with a comment like 
`# use x & y`